pkg/cache/clusterqueue_test.go

```diff
@@ -592,31 +599,33 @@ func TestClusterQueueUpdateWithAdmissionCheck(t *testing.T) {
 
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
-			cache := New(utiltesting.NewFakeClient())
-			cq, err := cache.newClusterQueue(cq)
-			if err != nil {
-				t.Fatalf("failed to new clusterQueue %v", err)
-			}
+			for _, cq := range cqs {
```

Comment:
following the DAMP not DRY principle, I propose that you instead add a field `cq` into the test cases struct, add more cases and remove this loop.

That way, we can clearly see what is the input cq and the expected output.

https://abseil.io/resources/swe-book/html/ch12.html#:~:text=Tests%20and%20Code%20Sharing%3A%20DAMP%2C%20Not%20DRY

Reply:
Fixed